tests/test_scheduler.py

```diff
@@ -0,0 +1,105 @@
+import os
+from unittest.mock import Mock, patch  # noqa: F401
+
+import pytest
+from mock import MagicMock
+
+from noisepy.seis.constants import AWS_BATCH_JOB_ID
+from noisepy.seis.scheduler import AWSBatchArrayScheduler, SingleNodeScheduler
+
+
+# Single Node Scheduler
+def test_single_node_scheduler_get_indices():
+    single_node_scheduler = SingleNodeScheduler()
+    items = ["a", "b", "c", "d"]
+    expected_indices = [0, 1, 2, 3]
+    indices = single_node_scheduler.get_indices(items)
+
+    assert indices == expected_indices
+
+
+def test_synchronize():
+    scheduler = SingleNodeScheduler()
+    result = scheduler.synchronize()
+
+    assert result is None
+
+
+# AWS Batch Array Scheduler
+
+
+# get_array_size test
+@patch("boto3.client")
+def test_get_array_size(mock_boto3_client):
+    boto3_response = {"jobs": [{"arrayProperties": {"size": 10}}]}
+
+    # Test 1 - valid job id format
+    # Mock the AWS_BATCH_JOB_ID environment variable with a valid format
+    os.environ["AWS_BATCH_JOB_ID"] = "job_id:123"
+    mock_boto3_client.return_value.describe_jobs.return_value = boto3_response
+    array_size = AWSBatchArrayScheduler._get_array_size()
+    assert array_size == 10
+
+    # Test 2 : Invalid job_id format
+    os.environ["AWS_BATCH_JOB_ID"] = "invalid_format"
+    with pytest.raises(ValueError) as e:
+        AWSBatchArrayScheduler._get_array_size()
+    assert str(e.value) == f"{AWS_BATCH_JOB_ID} is not in the expected format for an array job: 'invalid_format'"
```

Comment:
I suggest doing something like this:
```
assert "invalid_format" in str(e.value)
```
The reason is that having the full message has a minor drawbacks: It duplicates the error message between the code and the test and violates the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). This means that it the message is changed even a little bit, the test will break. 

By just asserting the presence of the `invalid_format` you remove the duplication and reduce the changes the test will need to be updated. But you still validate the main behavior, which is that there will be a ValueError about the 'invalid_format`.

You can do something similar with the other assertions below. E.g. `assert "123" in str(e.value)`